cloudinstall/charms/__init__.py

```diff
@@ -364,15 +360,32 @@ def watch_deploy(self):
                 self.ui.status_error_message(msg)
             time.sleep(10)
 
+    def filter_valid_relations(self):
+        deployed_charms_names = [x.charm_name for x in self.deployed_charms]
+        valid_relations = []
+        for c in self.deployed_charms:
+            for rel_a, rel_b in c.related:
+                rel_a_name = rel_a.split(":")[0]
+                rel_b_name = rel_b.split(":")[0]
+                if (rel_a_name and rel_b_name) in deployed_charms_names:
```

Comment:
I think this line doesn't do what you want, but it doesn't fail either! holy cow:
```

>>> l = ["B", "C"]
>>> ("A" and "B") in l
True
>>> "WHAT?"
'WHAT?'
>>> ("A" and "B")
'B'
>>> ("A" in l and "B" in l)
False
>>> 
```